File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution(object):
    def romanToInt(self, s):
        rom_num = { 
            "I": 1, 
            "V": 5, 
            "X": 10, 
            "L": 50, 
            "C": 100, 
            "D": 500, 
            "M": 1000
        }

        total = 0
        for i in range(len(s)-1):  
            if  rom_num[s[i]] < rom_num[s[i+1]]:
                total -= rom_num[s[i]] 

            else:
                total += rom_num[s[i]]

        return total + rom_num[s[-1]]

File: 0013-roman-to-integer/0013-roman-to-integer.java
class Solution {
    public int romanToInt(String s) {
        Map<Character, Integer> key = new HashMap<>();
        
        key.put('I', 1);
        key.put('V', 5);
        key.put('X', 10);
        key.put('L', 50);
        key.put('C', 100);
        key.put('D', 500);
        key.put('M', 1000);

        int ans = 0, i = 0;

        for(i = 0; i < s.length(); i++)
        {
            if (key.get(s.charAt(i)) < key.get(s.charAt(i+1)))
            {
                ans -= key.get(s.charAt(i));
            }
            else
            {
                ans += key.get(s.charAt(i));
            }
        }
        
        return ans;
    }
}